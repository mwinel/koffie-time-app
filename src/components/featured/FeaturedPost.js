import React, { Component } from "react";
import { Link } from "react-router-dom";
import { PostConsumer } from "../../context";
import Avatar from "../Avatar";
import CreateDate from "../Date";

class FeaturedPost extends Component {
  render() {
    const {
      title,
      image,
      user,
      avatar,
      created_on,
      read_time
    } = this.props.post;
    const styles = {
      backgroundImage: `linear-gradient(180deg, rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url(${image})`,
      backgroundSize: "cover",
      backgroundRepeat: "no-repeat",
      backgroundPosition: "50% 50%",
      height: "350px"
    };
    return (
      <PostConsumer>
        {value => (
          <>
            <div className="col-sm-7">
              <Link to={"/post"}>
                <div
                  className="featured-post-image-wrapper pb-5"
                  style={styles}
                ></div>
              </Link>
            </div>
            <div className="col-sm-5">
              <div className="pb-2">
                <CreateDate created_on={created_on} />{" "}
                <span className="dot-divider">&#183;</span>{" "}
                <small>{read_time}</small>
              </div>
              <Link to={"/post"}>
                <h2 className="card-title font-weight-bold">{title}</h2>
              </Link>
              <div className="flex flex-row flex-nowrap">
                <div className="py-2">
                  <Avatar avatar={avatar} />
                  <small className="username">
                    <Link to={"/"}>@{user}</Link>
                  </small>
                </div>
              </div>
            </div>
          </>
        )}
      </PostConsumer>
    );
  }
}

export default FeaturedPost;
