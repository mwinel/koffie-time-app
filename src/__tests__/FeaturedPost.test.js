import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { BrowserRouter } from "react-router-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import FeaturedPost from "../components/featured/FeaturedPost";
import value from "../testHelpers";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders featured-post component", () => {
  act(() => {
    render(
      <BrowserRouter>
        <FeaturedPost {...value} />
      </BrowserRouter>,
      container
    );
  });
  expect(container.innerHTML).toBeDefined();
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
