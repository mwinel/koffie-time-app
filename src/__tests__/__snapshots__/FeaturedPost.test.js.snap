// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders featured-post component 1`] = `
"<div class=\\"col-sm-7\\"><a href=\\"/post\\">
    <div class=\\"featured-post-image-wrapper pb-5\\" style=\\"background-size: cover; background-repeat: no-repeat; background-position: 50% 50%; height: 350px;\\"></div>
  </a></div>
<div class=\\"col-sm-5\\">
  <div class=\\"pb-2\\"><small>undefined NaN </small> <span class=\\"dot-divider\\">Â·</span> <small></small></div><a href=\\"/post\\">
    <h2 class=\\"card-title font-weight-bold\\"></h2>
  </a>
  <div class=\\"flex flex-row flex-nowrap\\">
    <div class=\\"py-2\\">
      <div class=\\"author-avatar mr-2\\" style=\\"background-size: cover; background-repeat: no-repeat; background-position: 50% 50%; width: 36px; height: 36px;\\"></div><small class=\\"username\\"><a href=\\"/\\">@</a></small>
    </div>
  </div>
</div>"
`;
